src/components/tables/EntityTable.tsx

```diff
@@ -53,178 +54,118 @@ interface Props {
-
-                        const connectorInfo: ConnectorInfo | undefined =
-                            connectors.find(
-                                (connector) =>
-                                    connector.image_name ===
-                                    publication.connector_image_name
-                            );
-
-                        return {
-                            deploymentStatus: 'ACTIVE',
-                            name: catalogNamespace.substring(
-                                catalogNamespace.lastIndexOf('/') + 1,
-                                catalogNamespace.length
-                            ),
-                            catalogNamespace,
-                            connectorType: connectorInfo
-                                ? connectorInfo.detail
-                                : 'Unknown',
-                            dateCreated,
-                        };
-                    }
-                );
-
-                setUnfilteredEntities(formattedPublication);
-                setTableState({ status: TableStatuses.DATA_FETCHED });
-            } else {
-                setTableState({ status: TableStatuses.NO_EXISTING_DATA });
-            }
-        }
-    }, [publications, connectors]);
+    const [pagination, setPagination] = useState<{ from: number; to: number }>(
+        getPagination(page, rowsPerPage)
+    );
 
-    useEffect(() => {
-        const subscription = supabaseClient
-            .from(TABLES.LIVE_SPECS)
-            .on('*', async (payload) => {
-                if (payload.new.spec_type === 'capture') {
-                    setPublications((_publications) =>
-                        _publications
-                            ? [..._publications, payload.new]
-                            : [payload.new]
+    const liveSpecQuery = useQuery<LiveSpecQuery>(
+        TABLES.LIVE_SPECS,
+        {
+            columns: LIVE_SPECS_QUERY,
+            count: 'exact',
+            filter: (query) => {
+                let queryBuilder = query;
+
+                // TODO (supabase) Change to text search? https://supabase.com/docs/reference/javascript/textsearch
+                if (searchQuery) {
+                    queryBuilder = queryBuilder.like(
+                        'catalog_name',
+                        `%${searchQuery}%`
                     );
                 }
-            })
-            .subscribe();
 
-        return () => {
-            supabaseClient.removeSubscription(subscription).catch(() => {});
-        };
-    }, [supabaseClient]);
+                return queryBuilder
+                    .order(columnToSort, {
+                        ascending: sortDirection === 'asc',
+                    })
+                    .range(pagination.from, pagination.to)
+                    .eq('spec_type', 'capture');
+            },
+        },
+        [sortDirection, columnToSort, searchQuery, rowsPerPage, pagination]
```

Comment:
Handles pagination, sorting, searching. So we have to pass those 4 deps to tell the hook when to run another query.